plugins/steelseries/fu-steelseries-fizz-tunnel.c

```diff
@@ -0,0 +1,323 @@
+/*
+ * Copyright (C) 2022 Gaël PORTAY <gael.portay@collabora.com>
+ *
+ * SPDX-License-Identifier: LGPL-2.1+
+ */
+
+#include "config.h"
+
+#include "fu-steelseries-firmware.h"
+#include "fu-steelseries-fizz-tunnel.h"
+
+struct _FuSteelseriesFizzTunnel {
+	FuDevice parent_instance;
+};
+
+G_DEFINE_TYPE(FuSteelseriesFizzTunnel, fu_steelseries_fizz_tunnel, FU_TYPE_DEVICE)
+
+static gboolean
+fu_steelseries_fizz_tunnel_wait_for_reconnect_cb(FuDevice *device,
+						 gpointer user_data,
+						 GError **error)
+{
+	guint8 status;
+
+	if (!fu_steelseries_fizz_connection_status(fu_device_get_parent(device), &status, error)) {
+		g_prefix_error(error, "failed to get connection status: ");
+		return FALSE;
+	}
+	g_debug("ConnectionStatus: %u", status);
+	if (status == 0) {
+		g_set_error(error, FWUPD_ERROR, FWUPD_ERROR_NOT_FOUND, "device is unreachable");
+		return FALSE;
+	}
+
+	/* success */
+	return TRUE;
+}
+
+static gboolean
+fu_steelseries_fizz_tunnel_wait_for_reconnect(FuDevice *device, guint delay, GError **error)
+{
+	return fu_device_retry_full(device,
+				    fu_steelseries_fizz_tunnel_wait_for_reconnect_cb,
+				    delay / 1000,
+				    1000,
+				    NULL,
+				    error);
+}
+
+static gboolean
+fu_steelseries_fizz_tunnel_open(FuDevice *device, GError **error)
+{
+	return fu_device_open(fu_device_get_parent(device), error);
+}
+
+static gboolean
+fu_steelseries_fizz_tunnel_close(FuDevice *device, GError **error)
+{
+	return fu_device_close(fu_device_get_parent(device), error);
+}
```

Comment:
I think you can delete this if you set `FU_DEVICE_INTERNAL_FLAG_USE_PARENT_FOR_OPEN`

Reply:
Will test it.

Reply:
Hum... got this

```
failed to get connection status: failed to do control transfer: Device 1038:1838 has not been opened
```
Needs to investigate.

Reply:
Ouch, it is an internal flag

Fixed by `fu_device_add_internal_flag(FU_DEVICE(self), FU_DEVICE_INTERNAL_FLAG_USE_PARENT_FOR_OPEN);`